### PR TITLE
Gives the xenoarch anomaly machines a nicer UI, alt and ctrl click.

### PR DESCRIPTION
**Pull request by PJB3005**
http://84.195.252.227/static/ShareX/2017/04/2017-04-16_18-24-24.mp4

### CODE REVIEW

**Comment by Gunosaguki** [comment]
Yes thank you for letting us know the progress of scanning

**Comment by Kurfursten** [comment]
Needs update wiki tag

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 1]
```diff
@@ -59,13 +61,11 @@
...
 		if(scan_process++ > target_scan_ticks)
 			FinishScan()
-		else if(temperature > 400)
+		else if(temperature > XENOARCH_MAX_TEMP)
 			src.visible_message("<span class='notice'>[bicon(src)] shuts down from the heat!</span>", 2)
 			scan_process = 0
-		else if(temperature > 350 && prob(10))
+		else if(temperature > XENOARCH_SAFETY_TEMP && prob(10))
 			src.visible_message("<span class='notice'>[bicon(src)] bleets plaintively.</span>", 2)
-			if(temperature > 400)
```
Comment (ComicIronic) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm:67]: This is why we need code coverage.

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 2]
```diff
@@ -218,3 +216,34 @@ obj/machinery/anomaly/proc/GetResultSpecifity(var/datum/geosample/scanned_sample
...
+
+	data["target_ticks"] = target_scan_ticks
+	data["scan_process"] = scan_process
+
+	data["beaker"] = !!held_container
+	if (held_container)
+		data["beaker_name"] = held_container.name
+		var/list/beaker_contents[0]
+		for(var/datum/reagent/R in held_container.reagents.reagent_list)
+			beaker_contents[++beaker_contents.len] = list(
```
Comment (ComicIronic) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm:234]: This is actually a sin. Do `beaker_contents.Add(list(list(...)))`

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 3]
```diff
@@ -218,3 +216,34 @@ obj/machinery/anomaly/proc/GetResultSpecifity(var/datum/geosample/scanned_sample
...
+	data["safety_temperature"] = XENOARCH_SAFETY_TEMP
+	data["temperature"] = temperature
+
+	data["target_ticks"] = target_scan_ticks
+	data["scan_process"] = scan_process
+
+	data["beaker"] = !!held_container
+	if (held_container)
+		data["beaker_name"] = held_container.name
+		var/list/beaker_contents[0]
```
Comment (ComicIronic) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm:232]: Why

The fixed size notation always annoys me - the lists aren't fixed size and never will be. This is also identical to `= list()`, and I prefer that much more for readability.

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm (hunk 2 above)
Comment (PJB3005) [reply code/modules/research/xenoarchaeology/machinery/analysis_base.dm:234]: No doing that is the sin.

**Pull request closed without merging**